0.13 docs update: tutorials (#4127)
garden-io/garden: docs/tutorials/your-first-project/1-initialize-a-project.md

We have one environment (`default`) and a single provider. We'll get back to this later.

> [//]: # (TODO: `garden create action` is still under construction, update this section when the command is ready)

Author: Agree, the command can wait. I'll change the tutorial to use explicit configuration code snippets for action definitions.

> [//]: # (TODO: `garden create action` is still under construction, update this section when the command is ready)

Author: Changed this in https://github.com/garden-io/garden/pull/4127/commits/1bc2439daa7c0ed83d226fe6bb25ea7e5e53c101

We have one environment (`default`) and a single provider. We'll get back to this later.

Next, let's create action configs for each of our two applications, starting with `backend`.

First we need to define `Build` and `Deploy` actions for the `backend` application. Let's use `container` action type.
Create an empty `backend.garden.yml` config file in the `backend` directory and add the following lines:

```yaml
kind: Build
name: backend
description: Backend service container image
type: container

---

kind: Deploy
name: backend
description: Backend service container
type: container

> # This defines an image to be used and refers the 'backend' Build action defined above

Reviewer: ```suggestion
# Reference to the Build action that builds the image to be deployed (defined above)
```

Now when I think this is a bit tricky to explain in a concise way. Best I could come up ^

build: backend

> # This block is necessary to deploy and expose the backend application

Reviewer: ```suggestion
# Action type specific config goes under the `spec` block
```

I think this is more useful

spec:
  healthCheck:
    httpGet:
      path: /hello-backend
      port: http
  ports:
    - name: http
      containerPort: 8080
      servicePort: 80
  ingresses:
    - path: /hello-backend
      port: http
```

Next, let's do the same for the `frontend` application:

> Create an empty `frontend.garden.yml` config file in the `frontend` directory and add the following lines:

Reviewer: ```suggestion
Create a `frontend.garden.yml` config file in the `frontend` directory and add the following lines:
```

```sh
kind: Build
name: frontend
description: Frontend service container image
type: container

---

kind: Deploy
name: frontend
description: Frontend service container
type: container

> # This defines an image to be used and refers the 'frontend' Build action defined above

Reviewer: I think this can be removed as it's explained above

build: frontend

> # Dependency section is used to ensure the deployment order. The frontend will be deployed after the backend.

Reviewer: ```suggestion
# Dependency section is used to specify action execution order. The frontend will be deployed after the backend is deployed.
# Dependency for the Build action is implicit.
```

dependencies:
  - deploy.backend

> # This block is necessary to deploy and expose the frontend application

Reviewer: I think this can be removed as it's explained above

Author: Removed in https://github.com/garden-io/garden/pull/4127/commits/e1230b742ace72516a30c9a2565fc9062f184f0a

spec:
  ports:
    - name: http
      containerPort: 8080
  healthCheck:
    httpGet:
      path: /hello-frontend
      port: http
  ingresses:
    - path: /hello-frontend
      port: http
    - path: /call-backend
      port: http
```

> Before deploying the application, you need to set up a local kubernetes cluster or connect to a remote cluster.

Reviewer: ```suggestion
Before deploying, you need to set up a local kubernetes cluster or connect to a remote cluster.
```
